aclk/mqtt.c

```diff
@@ -269,25 +276,13 @@ int _link_set_lwt(char *sub_topic, int qos)
         return 1;
     }
 
-    usec_t time_created_offset_usec = now_realtime_usec();
-    time_t time_created = time_created_offset_usec / USEC_PER_SEC;
-    time_created_offset_usec = time_created_offset_usec % USEC_PER_SEC;
-
-    char *msg_id = create_uuid();
-
-    snprintfz(
-        payload, 511,
-        "{ \"type\": \"disconnect\","
-        " \"msg-id\": \"%s\","
-        " \"timestamp\": %ld,"
-        " \"timestamp-offset-usec\": %llu,"
-        " \"version\": %d,"
-        " \"payload\": \"unexpected\" }",
-        msg_id, time_created, time_created_offset_usec, ACLK_VERSION);
-
-    freez(msg_id);
+    usec_t lwt_time = aclk_session_sec * USEC_PER_SEC + aclk_session_us + 1;
+    BUFFER *b = buffer_create(512);
+    aclk_create_header(b, "disconnect", NULL, lwt_time / USEC_PER_SEC, lwt_time % USEC_PER_SEC);
+    buffer_strcat(b, " \"payload\": \"unexpected\" }}");
+    buffer_free(b);
```

Comment:
This needs to move after the `mosquitto_will_set`

Reply:
Ok, this is the crash that @fracasula saw in the case that I could not test. Fixed.

Reply:
Fixed